### PR TITLE
2nd Assignment by 202502592(Hyukjin Chang)

### PR DESCRIPTION
**Pull request by hukjin1129**
# Assignment Submission

## Student Information
- **Student ID**: [Your Student ID]
- **Name**: [Your Full Name]
- **Assignment**: [Assignment Number]

## Changed Files

```
[FILE_TREE_PLACEHOLDER]
```

### Problem 1
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 2
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 3
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 4 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 5 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 6 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

## Testing
- [ ] I have run `python grader.py` locally
- [ ] All implemented problems pass their tests
- [ ] I have tested edge cases mentioned in the problem statements

## Additional Notes

---
**Note**: This PR will be automatically validated for:
- Correct directory structure
- Valid Python syntax
- PR title format: `Nth Assignment by {student_id} (Full Name)`
- File organization compliance

### CODE REVIEW

File: submissions/202502592/practice1/main.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return f"{self.name}: G{self.gold} S{self.silver} B{self.bronze}"
+
+    # Arithmetic overriding → 메달 합산
+    def __add__(self, other):
+        new_gold = self.gold + other.gold
+        new_silver = self.silver + other.silver
+        new_bronze = self.bronze + other.bronze
+        
+        return Country(self.name, new_gold, new_silver, new_bronze)
+        pass
```
Comment (hist0613) [inline submissions/202502592/practice1/main.py:20]: 필요없어진 pass는 지우는게 좋습니다

File: submissions/202502592/assignment2/problem2/textops/clean/filters.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      4) remove ASCII punctuation except apostrophes (') and hyphens (-)
+    """
+    # TODO: 구현하세요
+    # 힌트:
+    # 1) s.lower() - 소문자 변환
+    # 2) re.sub(r"\s+", " ", s) - 모든 연속 공백을 단일 공백으로
+    # 3) s.strip() - 앞뒤 공백 제거
+    # 4) string.punctuation에서 특정 문자 제외하고 제거
+    # 5) set 연산을 활용해서 keep = {"'", "-"}, 나머지는 제거
+    raise NotImplementedError
```
Comment (hist0613) [inline submissions/202502592/assignment2/problem2/textops/clean/filters.py:19]: 구현 미완료
